src/main_benchmark.cpp

```diff
@@ -218,8 +224,96 @@ void triggerInferenceInALoop(
     averageWholeLatency = std::accumulate(latenciesWhole.begin(), latenciesWhole.end(), 0) / (double(niterPerThread) * 1'000);
     averagePureLatency = std::accumulate(latenciesPure.begin(), latenciesPure.end(), 0) / (double(niterPerThread) * 1'000);
 }
+
+void triggerInferenceInALoop(
+    std::future<void>& startSignal,
+    std::promise<void>& readySignal,
+    const size_t niterPerThread,
+    size_t& wholeThreadTimeUs,
+    double& averageWholeLatency,
+    double& averagePureLatency,
+    OVMS_Server* server,
+    OVMS_InferenceRequest* request,
+    const std::string& inputName,
+    std::uint32_t elementsCount) {
+    OVMS_InferenceResponse* response{nullptr};
+    std::vector<uint64_t> latenciesWhole(niterPerThread);
+    std::vector<uint64_t> latenciesPure(niterPerThread);
+    readySignal.set_value();
+    startSignal.get();
+    auto workloadStart = std::chrono::high_resolution_clock::now();
+    size_t iter = niterPerThread;
+    while (iter-- > 0) {
+        float random = ((float) rand()) / (float) RAND_MAX;
+        std::vector<float> data(elementsCount, random);
+        OVMS_InferenceRequestInputSetData(request, inputName.c_str(), (const void*)data.data(), elementsCount * sizeof(float), OVMS_BUFFERTYPE_CPU, 0);
+        //InferenceRequest* ir = reinterpret_cast<InferenceRequest*>(request);
+        // stopSignal will be used with ctrl-c app stopping or with total requestCount
+        // stopSignal.wait_for(std::chrono::milliseconds(0));
+        auto iterationWholeStart = std::chrono::high_resolution_clock::now();
+        auto iterationPureStart = std::chrono::high_resolution_clock::now();
+        // aternatively we are changing request
+        OVMS_Inference(server, request, &response);
+        auto iterationPureEnd = std::chrono::high_resolution_clock::now();
+        OVMS_InferenceResponseDelete(response);
+        auto iterationWholeEnd = std::chrono::high_resolution_clock::now();
+        latenciesWhole[iter] = std::chrono::duration_cast<std::chrono::microseconds>(iterationWholeEnd - iterationWholeStart).count();
+        latenciesPure[iter] = std::chrono::duration_cast<std::chrono::microseconds>(iterationPureEnd - iterationPureStart).count();
+    }
+    auto workloadEnd = std::chrono::high_resolution_clock::now();
+    wholeThreadTimeUs = std::chrono::duration_cast<std::chrono::microseconds>(workloadEnd - workloadStart).count();
+    averageWholeLatency = std::accumulate(latenciesWhole.begin(), latenciesWhole.end(), 0) / (double(niterPerThread) * 1'000);
+    averagePureLatency = std::accumulate(latenciesPure.begin(), latenciesPure.end(), 0) / (double(niterPerThread) * 1'000);
+}
+
+void triggerInferenceInALoop(
```

Comment:
Do we have 3 functions doing most of the logic the same named the same?

Reply:
If no code can be shared between them then at least name them differently with reference to different modes.